File: reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        newHead = head
        while head and head.next:
            tmp = head.next
            head.next = tmp.next
            tmp.next = newHead
            newHead = tmp
        return newHead                             

File: reverse-linked-list/reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  let newHead = head;
  while(head && head.next) {
      let t = head.next;
      head.next = t.next;
      t.next = newHead;
      newHead = t;
  }
  return newHead;
};